### PR TITLE
stdlib: add option to run() to ignore exit code

### PR DESCRIPTION
**Pull request by mlichvar**
This will be useful in the chrony check actor (https://github.com/oamg/leapp-repository/pull/139).

### CODE REVIEW

File: leapp/libraries/stdlib/__init__.py [hunk 1]
```diff
@@ -140,7 +142,7 @@ def run(args, split=False, callback_raw=_logging_handler, env=None):
     try:
         create_audit_entry('process-start', {'id': _id, 'parameters': args, 'env': env})
         result = _call(args, callback_raw=callback_raw, env=env)
-        if result['exit_code'] != 0:
+        if not ignore_exit_code and result['exit_code'] != 0:
```
Comment (fernflower) [inline leapp/libraries/stdlib/__init__.py]: Maybe add warning at logging level? At least it won't be completely silent then

Comment (fernflower) [reply leapp/libraries/stdlib/__init__.py]: On second thought - probably not

**Comment by fernflower** [comment]
Can you please add a test for it?

File: leapp/libraries/stdlib/__init__.py [hunk 2]
```diff
@@ -119,7 +119,7 @@ def _logging_handler(fd_info, buffer):
         __write_raw(fd_info, buffer)
 
 
-def run(args, split=False, callback_raw=_logging_handler, env=None):
+def run(args, split=False, callback_raw=_logging_handler, env=None, ignore_exit_code=False):
```
Comment (vinzenz) [inline leapp/libraries/stdlib/__init__.py]:
```suggestion
def run(args, split=False, callback_raw=_logging_handler, env=None, checked=True):
```

File: leapp/libraries/stdlib/__init__.py [hunk 3]
```diff
@@ -131,6 +131,8 @@ def run(args, split=False, callback_raw=_logging_handler, env=None):
     :type split: bool
     :param callback_raw: Optional custom callback executed on raw data to print in console
     :type callback_raw: (fd: int, buffer: bytes) -> None
+    :param ignore_exit_code: Don't raise exception on a non-zero exit code
```
Comment (vinzenz) [inline leapp/libraries/stdlib/__init__.py]:
```suggestion
    :param checked: Raise an exception on a non-zero exit code, default True
```

File: leapp/libraries/stdlib/__init__.py [hunk 4]
```diff
@@ -131,6 +131,8 @@ def run(args, split=False, callback_raw=_logging_handler, env=None):
...
+    :type ignore_exit_code: bool
```
Comment (vinzenz) [inline leapp/libraries/stdlib/__init__.py]:
```suggestion
    :type checked: bool
```

File: leapp/libraries/stdlib/__init__.py (hunk 1 above)
Comment (vinzenz) [inline leapp/libraries/stdlib/__init__.py]:
```suggestion
        if checked and result['exit_code'] != 0:
```

Comment (mlichvar) [reply leapp/libraries/stdlib/__init__.py]: Ok. Makes sense.

**Comment by mlichvar** [comment]
> Can you please add a test for it?

Ok, I added two tests and changed the code to use the ``checked`` parameters

File: tests/scripts/test_stdlib.py [hunk 5]
```diff
@@ -24,6 +24,20 @@ def test_check_multiline_output_no_split():
     assert run(a_command, split=False)['stdout'] == u'This a multi-\nline No Split test!\n'
 
 
+def test_check_error():
+    a_command = ['false']
+    try:
```
Comment (fernflower) [inline tests/scripts/test_stdlib.py]: Please use pytest.raises for testing exception is raised, smth like 

```
with pytest.raises(CalledProcessError) as err:                                                  
    run(a_command, checked=True) 
```

https://docs.pytest.org/en/latest/assert.html

File: tests/scripts/test_stdlib.py [hunk 6]
```diff
@@ -24,6 +24,20 @@ def test_check_multiline_output_no_split():
     assert run(a_command, split=False)['stdout'] == u'This a multi-\nline No Split test!\n'
 
 
+def test_check_error():
+    a_command = ['false']
```
Comment (fernflower) [inline tests/scripts/test_stdlib.py]: I'd prefer not 'false' but something self-speaking like 'nosuchcommand'

File: tests/scripts/test_stdlib.py [hunk 7]
```diff
@@ -24,6 +24,20 @@ def test_check_multiline_output_no_split():
...
+    try:
+        run(a_command, checked=True)
+        assert False
+    except CalledProcessError:
+        pass
+
+
+def test_check_error_no_checked():
+    a_command = ['false']
```
Comment (fernflower) [inline tests/scripts/test_stdlib.py]: Same thing here

File: tests/scripts/test_stdlib.py (hunk 6 above)
Comment (mlichvar) [reply tests/scripts/test_stdlib.py]: I think we should be testing exit codes of existing commands here.

File: tests/scripts/test_stdlib.py (hunk 5 above)
Comment (mlichvar) [reply tests/scripts/test_stdlib.py]: Ok. Changed in the current patch.

File: leapp/libraries/stdlib/__init__.py [hunk 8]
```diff
@@ -131,6 +131,8 @@ def run(args, split=False, callback_raw=_logging_handler, env=None):
     :type split: bool
     :param callback_raw: Optional custom callback executed on raw data to print in console
     :type callback_raw: (fd: int, buffer: bytes) -> None
+    :param checked: Raise exception on a non-zero exit code, default True
```
Comment (vinzenz) [inline leapp/libraries/stdlib/__init__.py]:
```suggestion
    :param checked: Raise an exception on a non-zero exit code, default True
```

Comment (mlichvar) [reply leapp/libraries/stdlib/__init__.py]: Thanks.

File: tests/scripts/test_stdlib.py (hunk 6 above)
Comment (fernflower) [reply tests/scripts/test_stdlib.py]: makes sense

**Pull request merged**